cylc/flow/loggingutil.py

```diff
@@ -434,3 +434,25 @@ def patch_log_level(logger: logging.Logger, level: int = logging.INFO):
         logger.setLevel(orig_level)
     else:  # No need to patch
         yield
+
+
+def log_iterable(
+    items: Iterable,
+    header: str,
+    singular_header: str,
+) -> str:
+    """Log a list.
+
+    Args:
+        items: The iterable
+        singular_header: To use if list has length 1
+        header: To use otherwise
+    """
+    if len(items) == 0:
+        raise ValueError('This function does not take len==0 iterables')
+    elif len(items) == 1:
+        return singular_header.format(items[0])
+    else:
+        msg = header + '\n    * '
+        msg += '\n    * '.join(items)
+        return msg
```

Comment:
This function feels a bit over-complicated for a single use.

I think it should fail type checking as generators are iterables but are not supported by `len()`?

Reply:
It is over complicated for a single use. I think that this is code we've replicated in a number of places, and may want again in future . The idea is that this function is available for refactoring elsewhere and for future use cases.